onnxruntime/core/session/custom_ops.cc

```diff
@@ -248,6 +248,11 @@ common::Status CreateCustomRegistry(const std::vector<OrtCustomOpDomain*>& op_do
           .SetDomain(domain->domain_)
           .SinceVersion(1);
 
+      auto input_count = op->GetInputTypeCount(op);
+      for (size_t i = 0; i < input_count; i++) {
+        def_builder.InputMemoryType(op->GetInputMemoryType(op, i), i);
+      }
```

Comment:
Is this making it mandatory that a custom op author would need to have implemented this? I don't see any code in this PR that provides default behavior if they haven't done so.

Reply:
The default behavior is defined inside `CustomOpBase` as below. It can be found in this  PR as well at `include/onnxruntime/core/session/onnxruntime_cxx_api.h: Line 1020`.

```
  // Default implemention of GetInputMemoryType() that returns OrtMemTypeDefault
  OrtMemType GetInputMemoryType(size_t /*index*/) const {
    return OrtMemTypeDefault;
  }
```